File: jobs/2026-05-05__23-49-17/svix_fastapi_receiver__Y7JfzyH/artifacts/code/test_webhook.py
import unittest
from fastapi.testclient import TestClient
from main import app

class TestWebhook(unittest.TestCase):
    def setUp(self):
        self.client = TestClient(app)

    def test_missing_headers(self):
        response = self.client.post("/webhook", json={"test": "data"})
        self.assertEqual(response.status_code, 400)
        self.assertEqual(response.json()["detail"], "Missing Svix headers")

    def test_invalid_signature(self):
        headers = {
            "svix-id": "msg_p5jXN87S8vL5U69S",
            "svix-timestamp": "1614265330",
            "svix-signature": "v1,g0hM9L9v9639v999999999999999999999999999999="
        }
        response = self.client.post("/webhook", json={"test": "data"}, headers=headers)
        self.assertEqual(response.status_code, 400)
        self.assertEqual(response.json()["detail"], "Invalid signature")

if __name__ == "__main__":
    unittest.main()

File: jobs/2026-05-15__15-07-35/svix_nestjs_raw_body_parsing__7y96G2e/artifacts/reports/test-webhook.js
const { Webhook } = require('svix');
const http = require('http');

const secret = 'whsec_MfKQ9r8GffUjCNNdXbn5O6vTmQxOQ9XQ';
const payload = JSON.stringify({ test: 'data' });
const wh = new Webhook(secret);

async function test(headers, expectedStatus, label) {
  return new Promise((resolve, reject) => {
    const options = {
      hostname: 'localhost',
      port: 3000,
      path: '/webhook',
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        ...headers
      },
    };

    const req = http.request(options, (res) => {
      let body = '';
      res.on('data', (chunk) => body += chunk);
      res.on('end', () => {
        if (res.statusCode === expectedStatus) {
          console.log(`[PASS] ${label}: Received expected status ${res.statusCode}`);
          resolve();
        } else {
          console.log(`[FAIL] ${label}: Expected ${expectedStatus}, got ${res.statusCode}. Body: ${body}`);
          reject(new Error(`${label} failed`));
        }
      });
    });

    req.on('error', reject);
    req.write(payload);
    req.end();
  });
}

async function runTests() {
  const msgId = 'msg_123';
  const timestamp = new Date();
  const signature = wh.sign(msgId, timestamp, payload);

  // Test 1: Valid signature
  await test({
    'svix-id': msgId,
    'svix-timestamp': Math.floor(timestamp.getTime() / 1000).toString(),
    'svix-signature': signature,
  }, 200, 'Valid signature');

  // Test 2: Invalid signature
  await test({
    'svix-id': msgId,
    'svix-timestamp': Math.floor(timestamp.getTime() / 1000).toString(),
    'svix-signature': 'v1,invalid',
  }, 400, 'Invalid signature');

  // Test 3: Missing headers
  await test({
    'svix-id': msgId,
  }, 400, 'Missing headers');

  console.log('All tests passed!');
}

runTests().catch(err => {
  console.error(err);
  process.exit(1);
});
